contracts/zero-ex/contracts/src/features/TransformERC20.sol

```diff
@@ -316,20 +348,27 @@ contract TransformERC20 is
     /// @dev Executs a transformer in the context of `wallet`.
     /// @param wallet The wallet instance.
     /// @param transformation The transformation.
+    /// @param transformerDeployer The address of the transformer deployer.
     /// @param taker The taker address.
     /// @param callDataHash Hash of the calldata.
     function _executeTransformation(
         IFlashWallet wallet,
         Transformation memory transformation,
+        address transformerDeployer,
         address payable taker,
         bytes32 callDataHash
     )
         private
     {
+        // Derive the transformer address from the deployment nonce.
```

Comment:
so this means that if we ever replace the `transformerDeployer` address, all transformers deployed by the old deployer become inaccessible?

Reply:
Yup. We'll have to redeploy the existing ones.

Reply:
what was the rationale for this change?

Reply:
The code here or the deployer being replaceable?

Reply:
well the deployer being replaceable and the fact that old deployments become defunct once a deployer is replaced

Reply:
I decided to make it replaceable based off internal discussions and the uncertainty over how we want to manage the deployer at this stage. Previously we would have had to redeploy this entire feature. The fact that the transformers would die with the deployer is not new, because we always used the nonce to assert validity. This might actually be a good thing since it's a universal off switch for transformers if the deployer account is compromised. Ideally we won't be changing the deployer often, and we'll probably only have a handful of transformers early on.